pkg/scheduler/objects/application.go

```diff
@@ -114,10 +115,19 @@ func NewApplication(siApp *si.AddApplicationRequest, ugi security.UserGroup, eve
 		rejectedMessage:      "",
 		stateLog:             make([]*StateLogEntry, 0),
 	}
+
 	placeholderTimeout := common.ConvertSITimeout(siApp.ExecutionTimeoutMilliSeconds)
 	if time.Duration(0) == placeholderTimeout {
 		placeholderTimeout = defaultPlaceholderTimeout
 	}
+
+	var timeoutReached bool
+	timeoutReached, placeholderTimeout = app.adjustPlaceholderTimeout(placeholderTimeout, siApp)
+	if timeoutReached {
+		// Yunikorn was restarted and the placeholder timed out in the meantime
+		placeholderTimeout = time.Millisecond
```

Comment:
1 msec is basically instant timeout. There is a guard against the value 0 somewhere else, so this is the smallest allowed value.